### ISSUE TITLE
Server Crash on load

### ISSUE THREAD

**Issue by KaziArmada** [The-1-7-10-Pack/The-1.7.10-Pack#63]
So I have a small private server running the newest version of the pack. I was fiddling around, looking at recipes in NEI when I clicked on an electrum cover or something. The game promptly crashed, and upon checking it turned out that it crashed my server as well. 

Attempts to boot the server worked fine, but trying to reload ended up crashing it again. After speaking with a few of my players, I found that it only crashed the server when those players entered the specific area I had originally crashed in. Any attempts to join the game by anyone who had entered that area now crash the server instantly. 

Attached are all five crash reports I've gotten from this. Crash Prime is the initial crash, which happened while someone was in transit to the Twilight Forest. Crash 1 through 4 after are just me or someone else attempting to load into the area I had been in when I crashed.

https://www.dropbox.com/sh/3k6qq5xsyxmbqvz/AABPbTA1i-yIh7siZzIcSDSsa?dl=0

To clarify - No items were being placed into the world when this was happening. The worst that happened was, at the same time, one player was looking at NEI recipes and another was mid-transit to the Twilight Forest. 

Server Information
-----
CPU - AMD A-10 5800K APU, 3.8Ghz Quad Core
Graphics - Radion HD 7660D Integrated into the CPU
RAM - G-Skill Sniper 1866 PC3-14900, 8 Gigs. 
Motherboard - ASRock FM2A75 Pro4-M

**Comment by Flukethoughts** [comment]
Do you have all the mods enabled?

**Comment by KaziArmada** [comment]
Yes. The only changes made were a minor config change for Thaumcraft to disable taint spread and to disable flux, and adding Dynmap. No other changes have been made to the server mod-wise from the initial download.

**Comment by Flukethoughts** [comment]
UM why would you dare to add dynmap please post server specs

**Comment by KaziArmada** [comment]
>why would you dare to add dynmap

Because I like Dynmap.

>please post server specs

I don't see what the hardware spec's could have to do with this, given that from the 4 Crash logs besides Prime showing that this is some sort of Java error, but I can believe it might be related, so here. 

CPU - AMD A-10 5800K APU, 3.8Ghz Quad Core
Graphics - Radion HD 7660D Integrated into the CPU
RAM - G-Skill Sniper 1866 PC3-14900, 8 Gigs. 
Motherboard - ASRock FM2A75 Pro4-M

The server tends to run two minecraft server's at the same time, and save a bit of lag when it has to terrain gen, runs perfectly fine. Given that if I clone a new version of the server and flip it on that it's working fine, I'm pretty sure something in this specific world snapped in half for some odd reason.

**Comment by Flukethoughts** [comment]
The server does alot of terrain generation and can cause issues with corrupted chunks If you don't have the disk speed to meet up to the servers needs this will result in missing chunks. When you have all the mods enabled the server will always be overloaded. It doesn't matter if you have 5 or less players the server needs to be configured per system specs. Keep in mine minecraft is not multi-threaded.

**Comment by Flukethoughts** [comment]
From what I see u don't have the thread performance for dynmap being added tot he mod pack and you don't have the ram performance to supplement all the mods. Ram is a weird subject no matter how much you dedicated u have to have free ram for garbage collection. Also You might want to keep in mind you didn't post hdd specs. You can create a new world all you want but the problem will return back to you.

**Comment by KaziArmada** [comment]
I understand MC isn't multi-threaded, but you're really not giving me any answers here...just saying 'Clearly that isn't enough.' What IS a good baseline for hardware, since mine doesn't seem to be up to a spec you agree with? 

Also, I just want to point out this is the first time I've ever seen this sort of error without any real way for me to track it down myself and fix the issue...I've been using this server for almost two years at this point, and it's never had any issues keeping up with even heavily modded MC...it just requires careful use of resources, such as only letting Dynmap render at night, no new chunk gen without all players agreeing on it and letting it go, etc.

I'll admit it wouldn't hold up as a dedicated major public but I've never had any sort of issues like this, even with other 1.7.10 packs.

**Comment by KaziArmada** [comment]
So I tried fiddling with everything for a bit and, just to try something, pulled Dynmap from the list of mods. Loaded the server, and all player's that were stuck in the area are able to load and move around. I'm running a few more tests to try and figure out what happened, but it does look like Dynmap may have been related.

That said, I still don't understand WHY it was related, and how it crashed it just when players were in a specific area. I'm pretty sure that has nothing to do with the system's power and rather some sort of odd interaction I'm still attempting to nail down.

**Comment by Flukethoughts** [comment]
think of the thread as a glass of water that is constantly being filled. you have to drink all the water to play. Well If the pressure it fast entering the glass then what you are consuming parts of the game fall out of the glass.

**Comment by The-1-7-10-Pack** [comment]
Hey,
First of all - you have two kinds of crashes - two issues.
The "Prime" one is simply a TwilightForest bug, related as you said to Dynmap. You should report it there: https://github.com/Benimatic/twilightforest/issues
The second crash (that you posted 4 times) - I'm pretty sure it's EE3's fault, in the latest version of it they added some unfinished NEI addons that broke a lot of things. In the coming version of the modpack EE3 hopefully will be removed, so it should fix it then.
So, you can point out that you never saw such problems in modded mc but you can blame the modpack creator, there are things I can fix - but there are things I can't fix.

**Comment by KaziArmada** [comment]
>Well If the pressure it fast entering the glass then what you are consuming parts of the game fall out of the glass.

That still doesn't explain why only a single part failed. Moving around so the area around the affected house was NOT loaded resulted in zero issues. Soon as that one loaded? Entire server goes down and that player is essentially trapped until I pulled Dynmap. 

I refuse to believe it's simply 'The server can't handle the mod pack AND Dynmap' because that would result in issues beyond a simple, single fixed point.

**Comment by KaziArmada** [comment]
>The second crash (that you posted 4 times) - I'm pretty sure it's EE3's fault, in the latest version of it they added some unfinished NEI addons that broke a lot of things.

Really now? I'd never considered that, if only that nobody on the server has even considered EE3...but if it's doing goofy NEI things, I can see that for sure. 
Do you have any idea why it'd only trigger in a specific set of chunks which, as I'll again point out, had no EE3 items in it? And how said issue STOPPED happening when Dynmap was removed?

Quite odd...

**Comment by The-1-7-10-Pack** [comment]
You misunderstood me. Those are two issues - each with diffrent reasons.
NEI Problem - EE3.
Chunk/Area crash - TwilightForest & Dynmap.

**Comment by KaziArmada** [comment]
Ahh. Can you happen to figure out if a specific thing is triggering it? If I have to manually edit the map to kill an item at this point, I will.

**Comment by The-1-7-10-Pack** [comment]
It's not an item or a block it's a Biome.
As I understood you didn't really play in the Twilight Forest yet so I guess you can simply delete the twilight forest from the world folder.

**Comment by KaziArmada** [comment]
Tried that out, wiped out the Twilight Forest world...and it still ended up crashing. 

My character, whom I've done most of the testing with, is in the Overworld at this point. So unless you think the Twilight Portal itself is taking part in the crash, I'm unsure. I'd post more logs, but they read much the same as my initial ones.

**Comment by The-1-7-10-Pack** [comment]
Where and after what did you got the bugged chunk?

**Comment by KaziArmada** [comment]
The bugged chunk is in or around one of my players bases. I've actually managed to get into the world by importing it off the server and loading it up in single player, at which point it is apparently working fine. I'l trying to see if I can replicate the crash, but at least in the single-player loaded version it's working fine so far.

**Comment by KaziArmada** [comment]
I tried totally removing Twilight Forest since you said that was the catalyst behind the crash, and that FINALLY ended up getting me able to get the server loaded in MP without any issues. 

After this I replaced Twilight Forest and relaunched the server and the crash came back if I enter the afflicted area. I've been able to pretty reliably replicate this.

For now I've purged Twilight Forest from my mod list. I have a copy of the world with the aprox coordinates of the base where the crash happens if you want a copy to poke around, assuming you don't think it's just a one-off bug I caused by my fiddling. That said, I was never able to get the crash to show in single player, ONLY in MP.
